src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function add(...args) {
+    if (args.length === 0) {
+      const res = sum;
+
+      sum = 0;
+
+      return res;
+    }
+
+    sum += args.reduce((acc, val) => acc + val, 0);
```

Comment:
The use of `reduce` here is not needed as you will only ever have one argument. You can simply add the value directly to `sum`.